src/Plugin/resource/Resource.php

```diff
@@ -0,0 +1,303 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFieldDefinitions() {
+    return $this->fieldDefinitions;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDataProvider() {
+    if (isset($this->dataProvider)) {
+      return $this->dataProvider;
+    }
+    return $this->dataProviderFactory();
+  }
+
+  /**
+   * Constructs a Drupal\Component\Plugin\PluginBase object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param ResourceFieldCollectionInterface $field_definitions
+   *   The field definitions.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ResourceFieldCollectionInterface $field_definitions) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->fieldDefinitions = $field_definitions;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process() {
+    $path = $this->getPath();
+
+    return ResourceManager::executeCallback($this->getControllerFromPath($path), array($path));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function contollersInfo() {
+    // Provide sensible defaults for the HTTP methods. These methods (index,
+    // create, view, update and delete) are not implemented in this layer but
+    // they are guaranteed to exist because we are enforcing that all restful
+    // resources are an instance of \RestfulDataProviderInterface.
+    return array(
+      '' => array(
+        // GET returns a list of entities.
+        Request::METHOD_GET => 'index',
+        Request::METHOD_HEAD => 'index',
+        // POST
+        Request::METHOD_POST => 'create',
+      ),
+      // We don't know what the ID looks like, assume that everything is the ID.
+      '^.*$' => array(
+        Request::METHOD_GET => 'view',
+        Request::METHOD_HEAD => 'view',
+        Request::METHOD_PUT => 'replace',
+        Request::METHOD_PATCH => 'update',
+        Request::METHOD_DELETE => 'remove',
+      ),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getControllers() {
+    $controllers = array();
+    foreach (static::contollersInfo() as $path => $method_info) {
+      $controllers[$path] = array();
+      foreach ($method_info as $http_method => $controller_info) {
+        $controllers[$path][$http_method] = $controller_info;
+        if (!is_array($controller_info)) {
+          $controllers[$path][$http_method] = array('callback' => $controller_info);
+        }
+      }
+    }
+
+    return $controllers;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function index($path) {
+    return $this->getDataProvider()->index();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function view($path) {
+    // TODO: Compare this with 1.x logic.
```

Comment:
Just a note to make sure that the end result of a view is the same in 7.x-1.x and 7.x-2.x.